### PR TITLE
feat: [CDE-333]: Optimize docker image pull; Decouple pool.yaml storage from external storage config

### PR DESCRIPTION
**Pull request by vikyathharekal**
1. Optimize docker image pull by pointing the docker root directory to the external storage
2. Decouple pool.yaml storage from external storage config

# Commit Checklist

Thank you for creating a pull request! To help us review / merge this can you make sure that your PR adheres as much as possible to the following.

## The Basics

If you are adding new functionality, please provide evidence of the new functionality.

### CODE REVIEW

**Review by n00bitax — APPROVED** [review]

File: internal/cloudinit/cloudinit.go [hunk 1]
```diff
@@ -249,6 +249,17 @@ echo "done starting lite engine server"
 groupadd docker
 mkdir -p /opt/gitspaceagent
 
+echo "Updating docker root dir"
+systemctl stop docker
+mkdir -p /mnt/disks/mountdevcontainer/docker
```
Comment (dhruv-harness) [inline internal/cloudinit/cloudinit.go:254]: If we add the docker folder like this, we need to update the path which will be used by the gitspace as the source for bind mount. We do not want the gitspace to access the docker data dir.
Do we have the capability to return the storage path from OVH VM?

**Review by harshjain12 — DISMISSED** [review]

File: internal/cloudinit/cloudinit.go (hunk 1 above)
Comment (vikyathharekal) [reply internal/cloudinit/cloudinit.go:255]: Creating /mnt/disks/mountdevcontainer/gitspace below. 
Gitspace agent has been updated to point to this dir

File: internal/cloudinit/cloudinit.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ type Params struct {
 	Tmate                  types.Tmate
 	IsHosted               bool
 	GitspaceAgentConfig    types.GitspaceAgentConfig
-	StorageIdentifier      string
+	StorageIdentifier      types.StorageConfig
```
Comment (vistaarjuneja) [inline internal/cloudinit/cloudinit.go:34]: Nit: StorageConfig?

File: types/types.go [hunk 3]
```diff
@@ -127,8 +127,20 @@ type GitspaceOpts struct {
...
+
 type GitspaceAgentConfig struct {
 	Secret      string `json:"secret"`
 	AccessToken string `json:"access_token"`
 	Ports       []int  `json:"ports"`
 }
+
+type StorageConfig struct {
+	CephPoolIdentifier string `json:"ceph_pool_identifier"`
+	StorageIdentifier  string `json:"storage_identifier"`
```
Comment (vistaarjuneja) [inline types/types.go:144]: Nit: could just be called Identifier since it's already in StorageConfig

**Review by vistaarjuneja — APPROVED** [review]

**Pull request merged**